In [34]:
# This Python 3 environment comes with many helpful analytics libraries installed
# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python
# For example, here's several helpful packages to load

import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)

# Input data files are available in the read-only "../input/" directory
# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory

import os
for dirname, _, filenames in os.walk('/kaggle/input'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using "Save & Run All" 
# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session

/kaggle/input/competitions/titanic/train.csv
/kaggle/input/competitions/titanic/test.csv
/kaggle/input/competitions/titanic/gender_submission.csv


In [35]:
# Missing Indicator

In [36]:
import pandas as pd
import numpy as np

In [37]:
df=pd.read_csv('/kaggle/input/competitions/titanic/train.csv',usecols=['Age','Fare','Survived'])

In [38]:
df

,Survived,Age,Fare
0,0,22.0,7.2500
1,1,38.0,71.2833
2,1,26.0,7.9250
3,1,35.0,53.1000
4,0,35.0,8.0500
...,...,...,...
886,0,27.0,13.0000
887,1,19.0,30.0000
888,0,NaN,23.4500
889,1,26.0,30.0000


In [39]:
df.isnull().sum()

Survived      0
Age         177
Fare          0
dtype: int64

In [40]:
# First Fill missing Values using Mean

In [41]:
from sklearn.model_selection import train_test_split
from sklearn.impute import SimpleImputer


In [42]:
x=df.drop(columns=['Survived'])
y=df['Survived']

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2)

In [43]:
si=SimpleImputer(strategy='mean')

x_train_tras=si.fit_transform(x_train)
x_test_trans=si.transform(x_test)

In [44]:
from sklearn.linear_model import LogisticRegression

lr=LogisticRegression()
lr.fit(x_train_tras,y_train)

LogisticRegression()

In [49]:
from sklearn.metrics import accuracy_score
y_pred=lr.predict(x_test_trans)
print('Accuracy Score',accuracy_score(y_test,y_pred))

Accuracy Score 0.6927374301675978


# Using Missing Indictor

In [51]:
s=SimpleImputer(add_indicator=True)

x_train_miss=s.fit_transform(x_train)
x_test_miss=s.transform(x_test)


In [53]:
clf=LogisticRegression()
clf.fit(x_train_miss,y_train)

LogisticRegression()

In [57]:
pred=clf.predict(x_test_miss)
print('Accuracy_Score',accuracy_score(y_test,pred))

Accuracy_Score 0.7039106145251397


### Now see before Applying Missing Indictor Model Accuracy Was -> 0.69
### And after applying Missing Indictor Model Accuracy Is -> 0.70         
### So increase the model accuracy by 1